### ISSUE TITLE
Optimize focus behavior of Checkbox Wrapper

### ISSUE THREAD

**Issue by denyo** [porsche-design-system/porsche-design-system#2190]
Scope
---
Currently our `p-checkbox-wrapper` behaves different than a native `<input type="checkbox">` when wrapped within a `label`. When clicking our label, the `body` gets focused, while clicking the `label` of the native element focuses the `input`. This should be aligned.

Notes  
---
- https://stackblitz.com/edit/29nnpa?file=index.html
- Maybe this behaviour also affects other PDS form elements like Textfield, Textarea, Radio Button, Select, etc.

Acceptance criteria
---
- [ ] When clicking on the label of a PDS form element component then the `document.activeElement` should be the form element itself and not the body element

Subtasks
---
- [ ] Task
